### PR TITLE
Update how parameters, inital people in each compartment and contact …

### LINKED ISSUE #78: Correct python examples
Update python examples to new syntax how parameters, contact patterns and initial compartment populations are defined.

### PR DESCRIPTION
**Pull request by lelange**
…patterns are defined. Open questions are marked with TODO.

Closes #78 

## Merge Request - GuideLine Checklist 

**Guideline** to check code before resolve WIP and approval, respectively.
As many checkboxes as possible should be ticked.

### Checks by code author:
* [x] There is at least one issue associated with the pull request.
* [x] The branch follows the naming conventions as defined in the [git workflow](git-workflow).
* [ ] New code adheres with the [coding guidelines](coding-guidelines)
* [ ] Tests for new functionality has been added
* [x] A local test was successful
* [ ] There is appropriate **documentation** of your work. (use doxygen style comments)
* [ ] If new third party software is used, did you pay attention to its license? Please remember to add it to the wiki after successful merging.
* [ ] If new mathematical methods or epidemiological terms are used, has the glossary been updated ? Did you provide further documentation ?
 is present or referenced. Please provide your references.
* [ ] The following questions are addressed in the documentation*:  Developers (what did you do?, how can it be maintained?), For users (how to use your work?), For admins (how to install and configure your work?)
* For documentation: Please write or update the Readme in the current working directory!

### Checks by code reviewer(s):
* [ ] Is the code clean of development artifacts e.g., unnecessary comments, prints, ...
* [ ] The ticket goals for each associated issue are reached or problems are clearly addressed (i.e., a new issue was introduced).
* [ ] There are appropriate **unit tests** and they pass.
* [ ] The git history is clean and linearized for the merge request.
* [ ] Coverage report for new code is acceptable.

### CODE REVIEW

**Comment by mknaranja** [comment]
Where does "…patterns are defined. Open questions are marked with TODO." come from?

File: pycode/examples/secir_groups.py [hunk 1]
```diff
@@ -52,55 +54,54 @@ def run_secir_groups_simulation():
...
-        model.parameters.probabilities[i].set_infection_from_contact(1.0)
-        model.parameters.probabilities[i].set_carrier_infectability(0.67)
-        model.parameters.probabilities[i].set_asymp_per_infectious(0.09)
-        model.parameters.probabilities[i].set_risk_from_symptomatic(0.25)
-        model.parameters.probabilities[i].set_hospitalized_per_infectious(0.2)
-        model.parameters.probabilities[i].set_icu_per_hospitalized(0.25)
-        model.parameters.probabilities[i].set_dead_per_icu(0.3)
-        
+           
+        model.parameters.RelativeCarrierInfectability[AgeGroup(i)] = 0.67  #TODO: must this parameter be set?
```
Comment (mknaranja) [inline pycode/examples/secir_groups.py:84]: No, not necessarily. As all parameters, it has a default value. But it is better to explicitly call the setters for all our parameters in the example. Please remove comment.

File: pycode/examples/secir_groups.py [hunk 2]
```diff
@@ -52,55 +54,54 @@ def run_secir_groups_simulation():
...
-        
+           
+        model.parameters.RelativeCarrierInfectability[AgeGroup(i)] = 0.67  #TODO: must this parameter be set? 
+        model.parameters.InfectionProbabilityFromContact[AgeGroup(i)] = 1.0
+        model.parameters.AsymptoticCasesPerInfectious[AgeGroup(i)] = 0.09  # 0.01-0.16
+        model.parameters.RiskOfInfectionFromSympomatic[AgeGroup(i)] = 0.25  # 0.05-0.5
+        model.parameters.HospitalizedCasesPerInfectious[AgeGroup(i)] = 0.2  # 0.1-0.35
+        model.parameters.ICUCasesPerHospitalized[AgeGroup(i)] = 0.25  # 0.15-0.4
+        model.parameters.DeathsPerHospitalized[AgeGroup(i)] = 0.3  # 0.15-0.77
+        #TODO what about MaxRiskOfInfectionFromSympomatic?
```
Comment (mknaranja) [inline pycode/examples/secir_groups.py:91]: Please just set it as a dummy to twice the RiskOfInfectionFromSymptomatic. It is only used if the test-and-trace-capacity is used. You can add this as a comment. Good examples on the parameters that can be used should be get from the cpp tests: https://github.com/lelange/memilio/blob/main/cpp/tests/test_secir.cpp and https://github.com/lelange/memilio/blob/main/cpp/tests/test_secir_ageres.cpp

File: pycode/examples/secir_groups.py [hunk 3]
```diff
@@ -52,55 +54,54 @@ def run_secir_groups_simulation():
...
     # set contact frequency matrix
-    model.parameters.get_contact_patterns().cont_freq_mat[0].baseline = np.loadtxt(baseline_contact_matrix0) \
+    model.parameters.ContactPatterns.cont_freq_mat[0].baseline = np.loadtxt(baseline_contact_matrix0) \
         + np.loadtxt(baseline_contact_matrix1) + np.loadtxt(baseline_contact_matrix2) + np.loadtxt(baseline_contact_matrix3)
-    model.parameters.get_contact_patterns().cont_freq_mat[0].minimum = np.ones((num_groups, num_groups)) * 0
-
-    # Define Damping on Contacts
-    model.parameters.get_contact_patterns().cont_freq_mat.add_damping(secir.Damping(np.ones((num_groups, num_groups)) * 0.9, 30, 0, 0))
+    model.parameters.ContactPatterns.cont_freq_mat[0].minimum = np.ones((num_groups, num_groups)) * 0
+    model.parameters.ContactPatterns.cont_freq_mat.add_damping(Damping(coeffs = np.ones((num_groups, num_groups)) * 0.7, t = 30.0, level = 0, type = 0))
```
Comment (mknaranja) [inline pycode/examples/secir_groups.py:98]: Please stick to the same values in the examples (0.9 before, 0.7 now) if there isn't any reason to change.

File: pycode/examples/secir_simple.py [hunk 4]
```diff
@@ -43,54 +45,54 @@ def run_secir_simulation():
...
-    # Define Damping on Contacts
-    model.parameters.get_contact_patterns().cont_freq_mat.add_damping(secir.Damping(np.ones((num_groups, num_groups)) * 0.9, 30, 0, 0))
+    model.parameters.RelativeCarrierInfectability[A0] = 0.67 #TODO: must this parameter be set?
+    model.parameters.InfectionProbabilityFromContact[A0] = 1.0
+    model.parameters.AsymptoticCasesPerInfectious[A0] = 0.09  # 0.01-0.16
+    model.parameters.RiskOfInfectionFromSympomatic[A0] = 0.25  # 0.05-0.5
+    model.parameters.HospitalizedCasesPerInfectious[A0] = 0.2  # 0.1-0.35
+    model.parameters.ICUCasesPerHospitalized[A0] = 0.25  # 0.15-0.4
+    model.parameters.DeathsPerHospitalized[A0] = 0.3  # 0.15-0.77
+    #TODO what about MaxRiskOfInfectionFromSympomatic?
```
Comment (mknaranja) [inline pycode/examples/secir_simple.py:82]: see above

File: pycode/examples/secir_simple.py [hunk 5]
```diff
@@ -43,54 +45,54 @@ def run_secir_simulation():
...
+
+    #TODO how to do this: model.parameters.set_start_day(start_day + start_month * 30) # TODO: start day has to adapted more precisely!
```
Comment (mknaranja) [inline pycode/examples/secir_simple.py:84]: model.parameters.StartDay should do this.

File: pycode/examples/secir_simple.py [hunk 6]
```diff
@@ -43,54 +45,54 @@ def run_secir_simulation():
...
+
+    #TODO use "contacts = ContactMatrix(np.r_[0.5])" or differ between baseline and minimum?
```
Comment (mknaranja) [inline pycode/examples/secir_simple.py:86]: I don't understand.

**Review by mknaranja — COMMENTED** [review]
@lelange I hope that my answers were clear. 
@dabele Could you check once the changes are made and approve/merge if this is done still this week?

File: pycode/examples/secir_groups.py (hunk 3 above)
Comment (lelange) [reply pycode/examples/secir_groups.py:98]: I used the values that are in the testing example assuming it should be equal everywhere.

File: pycode/examples/secir_groups.py [hunk 7]
```diff
@@ -52,55 +54,56 @@ def run_secir_groups_simulation():
...
+        model.parameters.RelativeCarrierInfectability[AgeGroup(i)] = 0.67  
+        model.parameters.InfectionProbabilityFromContact[AgeGroup(i)] = 1.0
+        model.parameters.AsymptoticCasesPerInfectious[AgeGroup(i)] = 0.09  # 0.01-0.16
+        model.parameters.RiskOfInfectionFromSympomatic[AgeGroup(i)] = 0.25  # 0.05-0.5
+        model.parameters.HospitalizedCasesPerInfectious[AgeGroup(i)] = 0.2  # 0.1-0.35
+        model.parameters.ICUCasesPerHospitalized[AgeGroup(i)] = 0.25  # 0.15-0.4
+        model.parameters.DeathsPerHospitalized[AgeGroup(i)] = 0.3  # 0.15-0.77
+        model.parameters.MaxRiskOfInfectionFromSympomatic[AgeGroup(i)] = 0.5 # twice the value of RiskOfInfectionFromSymptomatic
+    
+    model.parameters.StartDay = start_day + start_month * 30 # TODO: start day has to adapted more precisely!
```
Comment (dabele) [inline pycode/examples/secir_groups.py:93]: `(date(start_year, start_month, start_day) - date(start_year, 1, 1)).days` should do it. Same below.

**Review by dabele — CHANGES_REQUESTED** [review]

**Review by dabele — APPROVED** [review]

**Pull request merged**